### ISSUE TITLE
First-Time User with multiple issues

### ISSUE THREAD

**Issue by anx-ag** [thedevs-network/kutt#408]
Hey there,

I tried to set this up on a Debian 10 server with apache as a reverse proxy but I'm having multiple issues here.
After fiddling around with Apache as a reverse proxy in front of it and SSL offloading etc. I just gave up on that for now because it always redirects me to pages that cannot be resolved, and sometimes I cannot see the static contents then (502 proxy error).

So for the time being, I'm trying to get it to work with http://localhost:3000 now (via SSH tunneling) and once that works, I'll have a look again at the offloading.

Question no. 1:
Is there a default user or a default password?
I've added my e-mail address to "ADMIN_EMAIL" but I'm not able to login with that account. I also did turn on registration temporarily, didn't help either.
When I use the "forgot password" function, no mails are being sent via my provided 127.0.0.1 postfix instance (which is otherwise working well and I'm also getting a red "an error occured" message) and when I try to register a new account, I'm also just getting "an error occured" and no further information.

Question no. 2:
Where are the logfiles and how can I turn on debug logging to see what's actually happening here?

I hope that I will be able to figure out all the other issues once I'm able to login and see some logging.

Thanks,
Alex

**Comment by MKRhere** [comment]
Hi Alex,

0) As for the reverse proxy, I haven't used Apache in many years, but will be glad to help you setup with nginx.

1) You'll need to create the account that you intend to use as admin user. If you're unable to register because you're having trouble with SMTP, you could try manually verifying by registering first, then running this SQL query:

```SQL
SELECT verification_token FROM users WHERE email='myuser@mydomain';
```

And visit this URL:

```text
http://localhost:$PORT/verify/$VERIFICATION_TOKEN
```

Replacing `PORT` with what you're using and `VERIFICATION_TOKEN` with the result from the previous step.

If the created user exists in `ADMIN_EMAIL` they will get admin privileges. I suspect you probably didn't get a reset mail because the user doesn't exist yet. If you still have issues with SMTP, it could possibly be raised as a separate issue.

3) The app does not write logfiles by itself, only logs to stdout. If you're running the app with systemd, you can retrieve the logs from `journalctl`.

**Comment by anx-ag** [comment]
Hey,

thanks for your reply.
The customer is already using Apache here, so port 443 is "blocked" by that; I will probably force him to switch to nginx as the offloader anyways, but trying to figure out how this tool works can be done in parallel without a working reverse proxy.

I did check the DB, there is one user entry in there for the e-mail address which is also part of the ADMIN_EMAIL configuration option, the verification_token although did not work, but that might be because it expired.

`# select verification_token,verification_expires from users;
          verification_token          |    verification_expires
--------------------------------------+----------------------------
 f31ab4c9-ea9f-4f69-9e94-ec6a073995d9 | 2020-12-04 14:34:55.568+00
(1 Zeile)
`

After moving the expiration date to tomorrow, it worked:

`update users set verification_expires = CURRENT_DATE + INTERVAL '1 day';
`

I could now successfully login and it seems to work.
How can I create additional users now? Is there a command line script to create them or is a record similar to the one I have in the "users" table for my account sufficient to create a new account? In the backend I cannot seem to find a way to do that.

Regarding the logs: I've started it via npm as mentioned in the instructions, so I think npm should forward all the output to the console, but there wasn't any, except for some warnings regarding a custom error and 404 page, but that's about it.

**Comment by MKRhere** [comment]
There isn't a command-line script at the moment; you can either register on the frontend or add another row to PostgreSQL as you noticed. You could create a shell script for the same if you're required to create a few users this way.

**Comment by anx-ag** [comment]
Alright, thanks - I've done that now and am adding users using a script now. So far, everything is running and this issue can be closed.

Best,
Alex

**Comment by MKRhere** [comment]
Cheers!
